test/src/tests-batchUploader.ts

```diff
@@ -305,157 +331,223 @@ describe('batch uploader', () => {
+            var batch3 = JSON.parse(window.fetchMock._calls[2][1].body);
+            var batch4 = JSON.parse(window.fetchMock._calls[3][1].body);
+            var batch5 = JSON.parse(window.fetchMock._calls[4][1].body);
+
+            // UAC event
+            expect(batch1.events.length, 'Batch 1: UAC event').to.equal(1);
+
+            // session start, AST
+            expect(
+                batch2.events.length,
+                'Batch 2: Session Start, AST'
+            ).to.equal(2);
+
+            // session end
+            expect(batch3.events.length, 'Batch 3: Session End').to.equal(1);
+
+            // session start, AST
+            expect(batch4.events.length, 'Batch 4: UAC event').to.equal(1);
+
+            // UAC event
+            expect(
+                batch5.events.length,
+                'Batch 5: Session Start, AST'
+            ).to.equal(2);
+
+            var batch1UAC = Utils.findEventFromBatch(
+                batch1,
+                'user_attribute_change'
+            );
+            batch1UAC.should.be.ok();
+
+            var batch2SessionStart = Utils.findEventFromBatch(
+                batch2,
+                'session_start'
+            );
+            var batch2AST = Utils.findEventFromBatch(
+                batch2,
+                'application_state_transition'
+            );
+
+            batch2SessionStart.should.be.ok();
+            batch2AST.should.be.ok();
+
+            var batch3SessionEnd = Utils.findEventFromBatch(
+                batch3,
+                'session_end'
+            );
+            batch3SessionEnd.should.be.ok();
+
+            var batch4UAC = Utils.findEventFromBatch(
+                batch4,
+                'user_attribute_change'
+            );
+            batch4UAC.should.be.ok();
+
+            var batch5SessionStart = Utils.findEventFromBatch(
+                batch5,
+                'session_start'
+            );
+            var batch5AST = Utils.findEventFromBatch(
+                batch5,
+                'application_state_transition'
+            );
+
+            batch5SessionStart.should.be.ok();
+            batch5AST.should.be.ok();
+
+            (typeof batch1.source_request_id).should.equal('string');
+            (typeof batch2.source_request_id).should.equal('string');
+            (typeof batch3.source_request_id).should.equal('string');
+            (typeof batch4.source_request_id).should.equal('string');
+            (typeof batch5.source_request_id).should.equal('string');
+
+            batch1.source_request_id.should.not.equal(batch2.source_request_id);
+            batch1.source_request_id.should.not.equal(batch3.source_request_id);
+            batch1.source_request_id.should.not.equal(batch4.source_request_id);
+            batch1.source_request_id.should.not.equal(batch5.source_request_id);
+            batch2.source_request_id.should.not.equal(batch3.source_request_id);
```

Comment:
Adding a line to distinguish between batch groups better
```suggestion
            batch1.source_request_id.should.not.equal(batch5.source_request_id);

            batch2.source_request_id.should.not.equal(batch3.source_request_id);
```